To short timeout for generateReport method.

[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**.
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository . to create an issue: https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jenkinsci_allure-2Dplugin_issues&d=DwID-g&c=FXJfUb1oWgygD0uNz-ujnA&r=0UmIrFabaQzxDSWUHSk4XGdSmWxwvG6dNIPErIqoA5w&m=Zk5f8fr-juxF9Wf6APpHK9wfIi4U6IWjAGkBMDwNxsY&s=-jDADsxOnLlZN00vHyYmwMmPaF6fVD-7aZdoIkbNPsU&e=
.
. Make sure you have a clear name for your issue. The name should start with a capital . letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins . - Support emoji in test descriptions
)

#### I'm submitting a ...
  * [x] bug report
  * [ ] feature request
  * [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?
For large amount of test results, allure framework generates report longer than 60 sec which cause timeout of ExecuteWatchdog. Allure framework process is killed and report is only partially generated. Exception is thrown:

[ERROR] Can't generate allure report data
org.apache.commons.exec.ExecuteException: Process exited with an error: 143 (Exit value: 143)

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. 
To reproduce the bug you have to generate large amount of data for allure framework. For example: web application tests which stores each send and received messages.


#### What is the expected behavior?
Timeout for AllureCommandline::generateReport should be configurable just like in case of AllureCommandline::serve.

#### What is the motivation / use case for changing the behavior?
Fix the malfunction of the main functionality.

#### Please tell us about your environment:

- Allure version:        2.7.0
- Test framework:        testng@6.14.3
- Allure adaptor:        allure-testng@2.6.0
- Generate report using: allure-maven@2.9
- Apache Maven     3.3.9
- maven-surefire-plugin    2.21.0

#### Other information

[//]: # (
. e.g. detailed explanation, stacktraces, related issues, suggestions . how to fix, links for us to have more context, eg. Stackoverflow, Gitter etc
)

--- a/src/main/java/io/qameta/allure/maven/AllureCommandline.java
+++ b/src/main/java/io/qameta/allure/maven/AllureCommandline.java
@@ -32,13 +32,13 @@ public class AllureCommandline {
     private final Path installationDirectory;
 
     public AllureCommandline(final Path installationDirectory, final String version) {
-        this(installationDirectory, version, null);
+        this(installationDirectory, version, DEFAULT_TIMEOUT);
     }
 
-    public AllureCommandline(final Path installationDirectory, final String version, final String timeout) {
+    public AllureCommandline(final Path installationDirectory, final String version, int timeout) {
         this.installationDirectory = installationDirectory;
         this.version = version;
-        this.timeout = timeout == null ? DEFAULT_TIMEOUT : Integer.valueOf(timeout);
+        this.timeout = timeout;
     }
 
     public int generateReport(List<Path> resultsPaths, Path reportPath) throws IOException {

--- a/src/main/java/io/qameta/allure/maven/AllureCommandline.java
+++ b/src/main/java/io/qameta/allure/maven/AllureCommandline.java
@@ -23,22 +23,22 @@ public class AllureCommandline {
 
     public static final String ALLURE_DEFAULT_VERSION = "2.0.1";
 
+    private static final int DEFAULT_TIMEOUT = 3600;
+
     private final String version;
 
-    private static final String SERVE_DEFAULT_TIMEOUT = "3600";
-
-    private final String serveTimeout;
+    private final int timeout;
 
     private final Path installationDirectory;
 
     public AllureCommandline(final Path installationDirectory, final String version) {
-        this(installationDirectory, version, SERVE_DEFAULT_TIMEOUT);
+        this(installationDirectory, version, null);
     }
 
-    public AllureCommandline(final Path installationDirectory, final String version, final String serveTimeout) {
+    public AllureCommandline(final Path installationDirectory, final String version, final String timeout) {
         this.installationDirectory = installationDirectory;
         this.version = version;
-        this.serveTimeout = serveTimeout == null ? SERVE_DEFAULT_TIMEOUT : serveTimeout;
+        this.timeout = timeout == null ? DEFAULT_TIMEOUT : Integer.valueOf(timeout);
     }
 
     public int generateReport(List<Path> resultsPaths, Path reportPath) throws IOException {
@@ -56,7 +56,7 @@ public class AllureCommandline {
         commandLine.addArgument("-o");
         commandLine.addArgument(reportPath.toAbsolutePath().toString(), true);
 
-        return execute(commandLine, 60);
+        return execute(commandLine, timeout);
     }
 
     public int serve(List<Path> resultsPaths, Path reportPath, String serveHost, Integer servePort) throws IOException {
@@ -78,7 +78,7 @@ public class AllureCommandline {
         for (Path resultsPath : resultsPaths) {
             commandLine.addArgument(resultsPath.toAbsolutePath().toString(), true);
         }
-        return execute(commandLine, Integer.valueOf(this.serveTimeout));
+        return execute(commandLine, timeout);
     }
 
     private void checkAllureExists() throws FileNotFoundException {

--- a/src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java
+++ b/src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java
@@ -78,7 +78,7 @@ public abstract class AllureGenerateMojo extends AllureBaseMojo {
      * Report timeout parameter in seconds.
      */
     @Parameter(property = "allure.report.timeout", defaultValue = "60")
-    protected String reportTimeout;
+    protected int reportTimeout;
 
     /**
      * The path to the allure.properties file

--- a/src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java
+++ b/src/main/java/io/qameta/allure/maven/AllureGenerateMojo.java
@@ -75,22 +75,10 @@ public abstract class AllureGenerateMojo extends AllureBaseMojo {
     private String reportDirectory;
 
     /**
-     * Serve timeout parameter in seconds.
+     * Report timeout parameter in seconds.
      */
-    @Parameter(property = "allure.serve.timeout")
-    protected String serveTimeout;
-
-    /**
-     * Serve host parameter.
-     */
-    @Parameter(property = "allure.serve.host")
-    protected String serveHost;
-
-    /**
-     * Serve port parameter.
-     */
-    @Parameter(property = "allure.serve.port", defaultValue = "0")
-    protected Integer servePort;
+    @Parameter(property = "allure.report.timeout", defaultValue = "60")
+    protected String reportTimeout;
 
     /**
      * The path to the allure.properties file
@@ -219,7 +207,7 @@ public abstract class AllureGenerateMojo extends AllureBaseMojo {
             Path reportPath = Paths.get(getReportDirectory());
 
             AllureCommandline commandline
-                    = new AllureCommandline(Paths.get(getInstallDirectory()), reportVersion);
+                    = new AllureCommandline(Paths.get(getInstallDirectory()), reportVersion, reportTimeout);
 
             getLog().info("Generate report to " + reportPath);
             commandline.generateReport(resultsPaths, reportPath);

--- a/src/main/java/io/qameta/allure/maven/AllureReportMojo.java
+++ b/src/main/java/io/qameta/allure/maven/AllureReportMojo.java
@@ -2,6 +2,7 @@ package io.qameta.allure.maven;
 
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
 
 import java.nio.file.Path;
 import java.nio.file.Paths;

--- a/src/main/java/io/qameta/allure/maven/AllureServeMojo.java
+++ b/src/main/java/io/qameta/allure/maven/AllureServeMojo.java
@@ -23,6 +23,24 @@ import static java.lang.String.format;
 public class AllureServeMojo extends AllureGenerateMojo {
 
     /**
+     * Serve timeout parameter in seconds.
+     */
+    @Parameter(property = "allure.serve.timeout", defaultValue = "3600")
+    private String serveTimeout;
+
+    /**
+     * Serve host parameter.
+     */
+    @Parameter(property = "allure.serve.host")
+    private String serveHost;
+
+    /**
+     * Serve port parameter.
+     */
+    @Parameter(property = "allure.serve.port", defaultValue = "0")
+    private Integer servePort;
+
+    /**
      * {@inheritDoc}
      */
     @Override

--- a/src/main/java/io/qameta/allure/maven/AllureServeMojo.java
+++ b/src/main/java/io/qameta/allure/maven/AllureServeMojo.java
@@ -26,7 +26,7 @@ public class AllureServeMojo extends AllureGenerateMojo {
      * Serve timeout parameter in seconds.
      */
     @Parameter(property = "allure.serve.timeout", defaultValue = "3600")
-    private String serveTimeout;
+    private int serveTimeout;
 
     /**
      * Serve host parameter.

